core-plugins/src/main/java/co/cask/hydrator/plugin/common/StructuredToOrcTransformer.java

```diff
@@ -68,18 +68,21 @@ public OrcStruct transform(StructuredRecord input) {
 
   public OrcStruct parseOrcSchema(Schema inputSchema) {
     if (schemaCache.containsKey(inputSchema)) {
-      return schemaCache.get(inputSchema);
-    }
-    StringBuilder builder = new StringBuilder();
-    try {
-      HiveSchemaConverter.appendType(builder, inputSchema);
-    } catch (UnsupportedTypeException e) {
-      throw new IllegalArgumentException(String.format("Not a valid Schema {}", inputSchema.toString()), e);
+      TypeDescription schema = schemaCache.get(inputSchema);
+      OrcStruct pair = (OrcStruct) OrcStruct.createValue(schema);
+      return pair;
+    } else {
+      StringBuilder builder = new StringBuilder();
+      try {
+        HiveSchemaConverter.appendType(builder, inputSchema);
+      } catch (UnsupportedTypeException e) {
+        throw new IllegalArgumentException(String.format("Not a valid Schema {}", inputSchema.toString()), e);
+      }
+      TypeDescription schema = TypeDescription.fromString(builder.toString());
+      OrcStruct pair = (OrcStruct) OrcStruct.createValue(schema);
```

Comment:
nit: a little cleaner to put this outside the if - else, and leave the condition just for getting schema.

another nit: 'pair' isn't a great name since its not a pair. you can just call it orcRecord or orcStruct or something like that.